server/game/cards/characters/03/aryastark.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../../drawcard.js');
+
+class AryaStark extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCharacterKilled: (event, player, card) => (
+                    this.controller === card.controller &&
+                    card.getFaction() === this.getFaction())
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    cardCondition: card => (
+                        card.location === 'play area' && 
+                        card.getType() === 'character' &&
+                        card.getStrength() <= 3),
+                    activePromptTitle: 'Select a character',
+                    waitingPromptTitle: 'Waiting for opponent to use ' + this.name,
+                    onSelect: (player, card) => this.onCardSelected(player, card)
+                });
+            }
+        });
+    }
+
+    onCardSelected(player, card) {
+        this.controller.sacrificeCard(this);
```

Comment:
This is relatively new (just merged today), but costs such as sacrificing the card should now be specified when you declare the reaction. See https://github.com/cryogen/throneteki/blob/master/docs/implementing-cards.md#paying-additional-costs-for-reactions-and-interrupts

So adding `cost: ability.costs.sacrificeSelf()` to the reaction declaration and removing this line will do the same thing.